### DISCUSSION TITLE
Mesh Offset or Alpha Wrapping to inside?

### DISCUSSION THREAD

**Discussion by lslzl3000** [CGAL/cgal#8533] [Q&A] [answered]
I want to extract a "shell" of a closed mesh, the simple idea is to erode an offset of the mesh/surface, and then do a boolean-operation to combine two parts.

But the `alpha_wrap_3` only add offset on `outside` of a mesh.
Any way to perform a `inside only offset` to common mesh? or any better way to do a `shell extraction`?

**Comment by MaelRL** [comment, marked as answer]
3D alpha wrapping by default wraps from outside. You can also wrap from inside if you set up seeds, see [this example](https://github.com/CGAL/cgal/blob/master/Alpha_wrap_3/examples/Alpha_wrap_3/wrap_from_cavity.cpp?rgh-link-date=2023-11-21T07%3A40%3A30Z).

**Comment by lslzl3000** [comment]
how about [extrude_mesh](https://doc.cgal.org/latest/Polygon_mesh_processing/group__PMP__meshing__grp.html#ga300c5eedd7d62bbd10ad2655405562f4), is this also works for mesh offsetting? And looks like it works for open meshes/surfaces.
